src/main/host/process.rs

```diff
@@ -63,6 +64,54 @@ impl Process {
         unsafe { &*self.memory_manager_ptr() }
     }
 
+    /// Register a descriptor and return its fd handle.
+    pub fn register_descriptor(&mut self, desc: CompatDescriptor) -> u32 {
```

Comment:
Maybe not in this MR, but it seems probably worth a type-safe wrapper for descriptor-numbers.

Reply:
I'll leave this for another PR, but maybe a [`RawFd`](https://doc.rust-lang.org/std/os/unix/io/type.RawFd.html), which is an `i32`, might be our best option.